### PR TITLE
Issue #66 Updates for Drupal 8.9

### LINKED ISSUE #66: Update Docker image to use Drupal 8.9.x

### PR DESCRIPTION
**Pull request by m4olivei**
Various updates to support Drupal 8.9. Resolves #66.

**Next steps:**

1. ~~Figure out the failure on `build_with_branch`~~
1. ~~Update the README~~
1. Merge this branch and tag `master` as `0.6.0-drupal89` the repo
1. Update the CicleCI config to automatically rebuild the tag daily.
1. When Docker has rebuilt and pushed the tag, update https://github.com/deviantintegral/drupal_tests_node_example/pull/6 to build against the new tag. Merge it.
1. Get back to https://github.com/Lullabot/media_mpx/pull/164

### CODE REVIEW

**Comment by m4olivei** [comment]
Not totally sure why the jobs are failing.

Also, currently adding the module under test depends on the [composer merge plugin](https://github.com/wikimedia/composer-merge-plugin) which was [removed in D9](https://www.drupal.org/project/drupal/issues/2912387). We'll need to find another way to do that.

**Comment by m4olivei** [comment]
The latest commit is better. I can get the `run-code-sniffer` job passing locally for instance. However, the image is still missing key composer dependencies, eg. `phpunit`, since the Drupal release downloaded from drupal.org via `duc` (https://github.com/deviantintegral/drupal-update-client) does not specify dev dependencies. Need to add in `drupal/core-dev`.

File: hooks/RoboFile.php
Comment (m4olivei) [inline hooks/RoboFile.php:79]: Turns out this is no longer used. There are probably other methods in here that are also no longer used.

File: setup.sh
Comment (m4olivei) [inline setup.sh:71]: Removing `behat/mink-selenium2-driver` and `drupal/coder` b/c they are included in `drupal/core-dev`.

File: setup.sh
Comment (m4olivei) [inline setup.sh:72]: Just switching to a tagged release here. Has been working in out project.

File: README.md
Comment (m4olivei) [inline README.md:82]: Removing `behat/mink-selenium2-driver` and `drupal/coder` b/c they are included in `drupal/core-dev`.

File: Dockerfile
Comment (m4olivei) [inline Dockerfile:125]: Removing `behat/mink-selenium2-driver` and `drupal/coder` b/c they are included in `drupal/core-dev`.

File: Dockerfile
Comment (m4olivei) [inline Dockerfile:123]: Just switching to a tagged release here. Has been working in out project.

File: Dockerfile
Comment (m4olivei) [inline Dockerfile]: This is the biggest change. What I was finding was that there were implicit assumptions in `Robofile.php` about the prescence of `wikimedia/composer-merge-plugin`. In 8.7, core required that out of the boc, which then also meant that core's `require-dev` dependencies came along for the ride and were installed as well. In 8.9, that's no longer the case. `wikimedia/composer-merge-plugin` is no longer used, and the core dev dependencies were not included. Given that, I decided to remove the use of the `duc` utility and use `composer create-project`. Here we start from `drupal/legacy-project` to give us a simplified file layout, the same as what we had. We then layer on `wikimedia/composer-merge-plugin` (we still want that for the purposes of merging the module under test's require and require-dev dependencies) and `drupal/core-dev` to get what we need to run tests.

**Comment by m4olivei** [comment]
@deviantintegral When you have a moment, would love you to take a first pass at this. I've found locally that it's running my tests for https://github.com/Lullabot/media_mpx (albeit with test errors, which as yet I _don't think_ are cased by the image).

Also, it looks like CircleCI is having a credentials problem for the jobs here? If you have any ideas, let me know.

If it looks good, I'll also update the README and CHANGELOG.

File: Dockerfile
Comment (deviantintegral) [reply Dockerfile]: TIL's about legacy-project, nice find! I imagine we'll have to change that later (given "legacy"), but that could always be when running against Drupal 9.

Comment (deviantintegral) [inline Dockerfile]: We may want to call out in the README that we're now up to Composer 2.

**Comment by deviantintegral** [comment]
We should probably update the VM image too - it's in a few places in the config: https://github.com/deviantintegral/drupal_tests/blob/master/.circleci/config.yml#L2-L5

For some reason the SSH Deploy Key was missing in the repo settings, but was still present in Circle. I re-added it at https://app.circleci.com/settings/project/github/deviantintegral/drupal_tests/ssh?return-to=%2F which should fix the builds.

**Comment by m4olivei** [comment]
At this point, I think we now need to update the test module https://github.com/deviantintegral/drupal_tests_node_example.

Also, I think that test module should have more bare copy of it's `composer.json` to be a true test. The `setup.sh` script is meant to require-dev the things that are needed for the module, however, those requirements are already set: 

https://github.com/deviantintegral/drupal_tests_node_example/blob/master/composer.json#L3-L10

Next step is to update that module to the latest 8.9 version and update its `composer.json` to be more representative.

File: .circleci/config.yml
Comment (m4olivei) [inline .circleci/config.yml:85]: Will need to remember to tag `drupal_tests` after this PR is merged, and then either manually run `build_and_push` or uncomment the cron schedule (eg. https://github.com/deviantintegral/drupal_tests/blob/master/.circleci/config.yml#L95-L104)

**Comment by m4olivei** [comment]
I'm also wondering about the database dump https://github.com/deviantintegral/drupal_tests/blob/master/drupal.sql.gz. Does that need to be updated here for D9?

**Comment by deviantintegral** [comment]
It's not required - the point of it is to skip the `site:install`, which is typically slower than importing a small db and running update hooks. But, if it's easy, go for it?

Oh, I wrote a script for this, so hopefully just running https://github.com/deviantintegral/drupal_tests/blob/master/export-drupal-db.sh would do it.

**Comment by m4olivei** [comment]
Some progress today! I've been building the image described by the `Dockerfile` on this branch and pushing it to https://quay.io/repository/m4olivei/drupal-8.9-test. I've also been updating the example module repo here https://github.com/deviantintegral/drupal_tests_node_example/pull/6 agasint that image, and just got most of the jobs to run green! The code sniffer job is still failing, but in an expected way. Turns out node module isn't totally written to Drupal coding standards :laughing: .

**Comment by m4olivei** [comment]
ITS GREEN!

giphy

**Comment by m4olivei** [comment]
The DB export command (https://github.com/deviantintegral/drupal_tests/blob/master/export-drupal-db.sh) didn't work. I'll file a follow up to debug and fix that. However I was able to pick apart the commands and run them manually to get a usable export.

File: Dockerfile
Comment (deviantintegral) [inline Dockerfile:102]: 👏

File: hooks/code-sniffer.sh
Comment (deviantintegral) [inline hooks/code-sniffer.sh:19]: This is great and should we document this support in the README? Or does drupal.org support this now too?

**Review by deviantintegral — APPROVED** [review]
Most excellent! Feel free to merge when you see it as ready. Thank you!

File: hooks/code-sniffer.sh
Comment (m4olivei) [reply hooks/code-sniffer.sh:19]: I don't think drupal.org supports it, no. I did this to get the node example module tests green, thought it would be useful for others. Will find a spot for it in the README, good call.

**Comment by m4olivei** [comment]
Well that's frustrating. The README change broke the jobs :rofl: :sob: . I'm going to keep going ignoring that failure, which doesn't seem related, and deal with it in a follow up if necessary.

File: .circleci/config.yml
Comment (m4olivei) [reply .circleci/config.yml:85]: This is done and the job succedded: 

https://app.circleci.com/pipelines/github/deviantintegral/drupal_tests/23/workflows/52200dba-6db9-4324-bd93-8a9b69688ca7/jobs/526
https://quay.io/repository/deviantintegral/drupal_tests?tab=tags

**Pull request merged**
